Index: lib/Parse/ParseStmtAsm.cpp
===================================================================
--- lib/Parse/ParseStmtAsm.cpp	(revision 242392)
+++ lib/Parse/ParseStmtAsm.cpp	(working copy)
@@ -477,7 +477,7 @@
   const llvm::Target *TheTarget = nullptr;
   bool UnsupportedArch =
       (ArchTy != llvm::Triple::x86 && ArchTy != llvm::Triple::x86_64);
-  if (UnsupportedArch) {
+  /*if (UnsupportedArch) {
     Diag(AsmLoc, diag::err_msasm_unsupported_arch) << TheTriple.getArchName();
   } else {
     std::string Error;
@@ -484,7 +484,7 @@
     TheTarget = llvm::TargetRegistry::lookupTarget(TT, Error);
     if (!TheTarget)
       Diag(AsmLoc, diag::err_msasm_unable_to_create_target) << Error;
-  }
+  }*/
 
   assert(!LBraceLocs.empty() && "Should have at least one location here");
 
